os/board/rtl8730e/src/rtl8730e_st7701.c

```diff
@@ -0,0 +1,198 @@
+
+struct st7701_config_s g_rtl8730_st7701_config_s = {
+	.init = rtl8730_st7701_lcd_init,
+	.reset = rtl8730_st7701_gpio_reset,
+	.cache_invalidate = rtl8730_st7701_cache_invalidate,
+	.lcd_reload = rtl8730_st7701_lcd_reload,
+	.lcd_enable = rtl8730_st7701_lcd_enable,
+	.lcd_layer_enable = rtl8730_st7701_lcd_layer_enable,
+	.lcd_put_area = rtl8730_st7701_lcd_put_area,
+};
+
+LCDC_TypeDef *pLCDC = LCDC;
+LCDC_InitTypeDef lcdc_init_struct;
+
+u8 *lcd_img_buffer1 = (u8 *)(DDR_BASE + LCDC_IMG_BUF_OFFSET2 + LCDC_IMG_BUF_SIZE);
+
+static void LcdcInitValues(struct lcd_data config)
+{
+	LCDC_StructInit(&lcdc_init_struct);
+	lcdc_init_struct.LCDC_ImageWidth = config.XPixels;
+	lcdc_init_struct.LCDC_ImageHeight = config.YPixels;
+	for (u8 idx = 0; idx < LCDC_LAYER_MAX_NUM; idx++) {
+		lcdc_init_struct.layerx[idx].LCDC_LayerImgFormat = LCDC_LAYER_IMG_FORMAT_RGB565;
+		lcdc_init_struct.layerx[idx].LCDC_LayerImgBaseAddr = (u32) lcd_img_buffer1;
+		lcdc_init_struct.layerx[idx].LCDC_LayerHorizontalStart = 1;
+		lcdc_init_struct.layerx[idx].LCDC_LayerHorizontalStop = config.XPixels;
+		lcdc_init_struct.layerx[idx].LCDC_LayerVerticalStart = 1;
+		lcdc_init_struct.layerx[idx].LCDC_LayerVerticalStop = config.YPixels;
+		lcdc_init_struct.layerx[idx].LCDC_LayerColorKeyingEn = ENABLE;
+	}
+}
+
+static u32 LCDCRgbHvIsr(void *Data)
+{
+	LCDC_TypeDef *LCDCx = (LCDC_TypeDef *) Data;
+	volatile u32 IntId = LCDC_GetINTStatus(LCDCx);
+
+	if (IntId & LCDC_BIT_LCD_LIN_INTS) {
+		LCDC_ClearINT(LCDCx, LCDC_BIT_LCD_LIN_INTS);
+	}
+	if (IntId & LCDC_BIT_DMA_UN_INTS) {
+		LCDC_ClearINT(LCDCx, LCDC_BIT_DMA_UN_INTS);
+	}
+	return 0;
+}
+
+static void rtl8730_st7701_lcd_init(void)
+{
+	rtl8730_st7701_lcd_layer_enable(0, true);
+	rtl8730_st7701_lcd_layer_enable(1, false);
+	rtl8730_st7701_lcd_layer_enable(2, false);
+
+	LCDC_Init(pLCDC, &lcdc_init_struct);
+	LCDC_DMAModeConfig(pLCDC, LCDC_LAYER_BURSTSIZE_4X64BYTES);
+	LCDC_DMADebugConfig(pLCDC, LCDC_DMA_OUT_DISABLE, 0);
+
+	LCDC_LineINTPosConfig(pLCDC, YRES * 4 / 5);
+	LCDC_INTConfig(pLCDC, LCDC_BIT_LCD_LIN_INTEN | LCDC_BIT_DMA_UN_INTEN, ENABLE);
+}
+
+static void rtl8730_st7701_gpio_reset(void)
+{
+	return;
+}
+
+static void rtl8730_st7701_cache_invalidate(u32 *buffer, int size)
+{
+	DCache_CleanInvalidate((u32) buffer, size);
+}
+
+static void rtl8730_st7701_lcd_reload(void)
+{
+	LCDC_TrigerSHWReload(pLCDC);
+}
+
+static void rtl8730_st7701_lcd_enable(void)
+{
+	rtl8730_st7701_enable_lcdc();
+}
+
+static void rtl8730_st7701_lcd_layer_enable(int layer, bool enable)
+{
+	if (enable) {
+		lcdc_init_struct.layerx[layer].LCDC_LayerEn = ENABLE;
+	} else {
+		lcdc_init_struct.layerx[layer].LCDC_LayerEn = DISABLE;
+	}
+}
+
+static void rtl8730_st7701_lcd_put_area(u8 *lcd_img_buffer, u32 x1, u32 y1, u32 x2, u32 y2)
+{
+	lcdc_init_struct.layerx[0].LCDC_LayerImgBaseAddr = (u32) lcd_img_buffer;
+
+	//Uncomment below lines in case LCD window size needs to be changed...
+	/*
```

Comment:
I'm not sure is this the only way to achieve change in window size, perhaps using some #if CONFIG..... to control?
Also, if the input arguments is not used, you should void it.
eg: `(void) x1; ......`